targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC55S69/TARGET_M33_S/device/TOOLCHAIN_ARMC6/startup_LPC55S69_cm33_core0.S

```diff
@@ -157,9 +134,6 @@ Reset_Handler   PROC
                 ;MSR     MSPLIM, R0
                 LDR     R0, =SystemInit
                 BLX     R0
-                MRS     R0, control    ; Get control value
-                ORR     R0, R0, #2     ; Select switch to PSP, which will be set by __user_initial_stackheap
-                MSR     control, R0
```

Comment:
Why is this code deleted. From the documentation this bit selects the stack pointer that is used.

Reply:
It's not clear to me what it was trying to do.

I was baulking at that operation being illegal - ARMv7-M says setting the SPSEL bit in Handler mode is reserved. But I see ARMv8-M says it's just ignored.

Still, I don't see what you're trying to achieve here. Any switch to Thread Mode + PSP will occur in the RTOS, and it will be dealing with SPSEL. I would expect all target startup code to remain in Handler mode + MSP.

Is there something I'm missing about the startup code for the secure side here?

Reply:
Okay, been staring at TF-M main repo for a while. I can see those lines are in its startup assembler files. A couple of them follow it by an attempt to set `SP` to the top of `ARM_LIB_STACK`, which seems erroneous, as we're in Handler mode, so we are not switching to PSP now.

I can't find `CONTROL_S.SPSEL` being manipulated anywhere else in the TF-M source, so I think the effect of those lines may be to make the secure side run with separate Process stack, when eventually a Thread Mode secure->non-secure gateway call happens. The PSP itself is actually set up elsewhere (`tfm_spm_db_init`?)

Not clear why this is placed in target-specific startup, rather than that `tfm_spm_db_init`, as it doesn't do anything immediate - maybe it's a remnant from that old failed attempt to initialise PSP in target startup code?

I can put it back, but I'd like to understand a bit better. That reference to `__user_initial_stackheap` is wrong...

Reply:
I have received this code from NXP TFM team. I checked with the TFM team, they said this implementation is based on the reference received for the MUSCA platform.

Reply:
We can change the comment to "/* Select switch to PSP */", I agree that is wrong and it has been corrected in the updated version of this file

Reply:
I've put the code back, adjusting just the comment.

Issue raised over on TF-M to query what's going on here:

https://developer.trustedfirmware.org/T565